vm/src/stdlib/io.rs

```diff
@@ -115,40 +128,50 @@ impl BufferedIO {
         self.cursor.position()
     }
 
-    fn readline(&mut self, size: OptionalOption<i64>, vm: &VirtualMachine) -> PyResult<Vec<u8>> {
-        let mut buf = Vec::new();
-        let mut left = size
-            .flatten()
-            .filter(|s| *s >= 0)
-            .and_then(|s| s.try_into().ok())
-            .unwrap_or(usize::MAX);
-        if left == 0 {
-            return Ok(buf);
-        }
-        loop {
-            let (done, used) = {
-                let mut available = self.cursor.fill_buf().map_err(|err| os_err(vm, err))?;
```

Comment:
`fill_buf` read more data from reading source when the buffer is empty.
In this case, buffer is always the cursor itself and it doesn't have its background buffer source.
So we don't need to use loop here unless we are feeding new data into cursor ourselves.

Reply:
Ah true, it will return all of the remaining data. Might be good to add a comment to that affect.